examples/chef/common/clusters/valve-configuration-and-control/chef-valve-configuration-and-control-delegate-impl.cpp

```diff
@@ -0,0 +1,165 @@
+                                              const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer)
+{
+    chip::Protocols::InteractionModel::Status ret = chip::Protocols::InteractionModel::Status::Success;
+
+    if (endpointId != 1)
+    {
+        ChipLogError(DeviceLayer, "Endpoint %d not supported for ValveConfigurationAndControl cluster.", endpointId);
+        return chip::Protocols::InteractionModel::Status::UnsupportedEndpoint;
+    }
+
+    chip::AttributeId attributeId = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::ValveConfigurationAndControl::Attributes::RemainingDuration::Id: {
+        uint32_t newVal = 0;
+        std::memcpy(&newVal, buffer, sizeof(uint32_t));
+        ChipLogProgress(DeviceLayer, "Setting RemainingDuration to %d", newVal);
+        DataModel::Nullable<uint32_t> aRemainingDuration(newVal);
+        CHIP_ERROR err = SetRemainingDurationExt(endpointId, aRemainingDuration);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(DeviceLayer, "Unable to write RemainingDuration");
+            ret = chip::Protocols::InteractionModel::Status::Failure;
+        }
+        break;
+    }
+    break;
+    default:
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedAttribute;
+        ChipLogError(DeviceLayer, "Unsupported Attribute ID: %d", static_cast<int>(attributeId));
+        break;
+    }
+
+    return ret;
+}
+
+chip::Protocols::InteractionModel::Status
+chefValveConfigurationAndControlReadCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                             const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer,
+                                             uint16_t maxReadLength)
+{
+    chip::Protocols::InteractionModel::Status ret = chip::Protocols::InteractionModel::Status::Success;
+
+    if (endpointId != 1)
+    {
+        ChipLogError(DeviceLayer, "Endpoint %d not supported for ValveConfigurationAndControl cluster.", endpointId);
+        return chip::Protocols::InteractionModel::Status::UnsupportedEndpoint;
+    }
+
+    chip::AttributeId attributeId = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::ValveConfigurationAndControl::Attributes::RemainingDuration::Id: {
+
+        DataModel::Nullable<uint32_t> duration;
+        CHIP_ERROR err = GetRemainingDurationExt(endpointId, duration);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(DeviceLayer, "Unable to read RemainingDuration");
+            ret = chip::Protocols::InteractionModel::Status::Failure;
+            break;
+        }
+        if (duration.IsNull())
+        {
+            std::memset(buffer, '\0', maxReadLength);
+        }
+        else
+        {
+            uint32_t val = duration.Value();
+            std::memcpy(buffer, &val, std::min(maxReadLength, static_cast<uint16_t>(sizeof(uint32_t))));
```

Comment:
```suggestion
            std::memcpy(buffer, &val, std::min<size_t>(maxReadLength, sizeof(uint32_t)));
```

feels more readable somehow.

Reply:
Yes, Done.